test/Passes/xsmm-combine.mlir

```diff
@@ -262,3 +262,112 @@ func.func @none_on_binary_add_bf16(%arg0: memref<256x128xbf16>) -> memref<256x51
+    %8 = xsmm.binary.dispatch add [32, 32, 32, 32, 32] flags = (bcast_col_in0) data_type = bf16
+    xsmm.binary add(data_type = bf16, %8, %0, %alloc_3, %alloc_3) : (i64, memref<32xbf16>, memref<32x32xbf16>, memref<32x32xbf16>) -> ()
+    %subview_4 = memref.subview %alloc_0[%arg1, %arg2, 0, 0] [1, 1, 32, 32] [1, 1, 1, 1] : memref<8x32x32x32xbf16> to memref<32x32xbf16, strided<[32, 1], offset: ?>>
+    %9 = xsmm.unary.dispatch relu [32, 32, 32, 32] flags = (none) data_type = bf16
+    xsmm.unary relu(data_type = bf16, %9, %alloc_3, %subview_4) : (i64, memref<32x32xbf16>, memref<32x32xbf16, strided<[32, 1], offset: ?>>) -> ()
+    memref.dealloc %alloc_3 : memref<32x32xbf16>
+  }
+  scf.forall (%arg1, %arg2) = (0, 0) to (256, 1024) step (32, 32) {
+    %6 = affine.apply affine_map<(d0) -> (d0 floordiv 32)>(%arg1)
+    %7 = affine.apply affine_map<(d0) -> (d0 floordiv 32)>(%arg2)
+    %subview = memref.subview %alloc_0[%6, %7, 0, 0] [1, 1, 32, 32] [1, 1, 1, 1] : memref<8x32x32x32xbf16> to memref<32x32xbf16, strided<[32, 1], offset: ?>>
+    %subview_3 = memref.subview %alloc[%arg1, %arg2] [32, 32] [1, 1] : memref<256x1024xbf16> to memref<32x32xbf16, strided<[1024, 1], offset: ?>>
+    %8 = xsmm.unary.dispatch identity [32, 32, 32, 1024] flags = (none) data_type = bf16
+    xsmm.unary identity(data_type = bf16, %8, %subview, %subview_3) : (i64, memref<32x32xbf16, strided<[32, 1], offset: ?>>, memref<32x32xbf16, strided<[1024, 1], offset: ?>>) -> ()
+  }
+  memref.dealloc %alloc_0 : memref<8x32x32x32xbf16>
+  memref.dealloc %alloc_1 : memref<8x32x32x32xbf16>
+  memref.dealloc %alloc_2 : memref<8x32x32x32xbf16>
+  return %alloc : memref<256x1024xbf16>
+}
+
+// CHECK-LABEL: func.func @forward(
+// CHECK: %[[ARG0:.*]]: memref<256x1024xbf16>) -> memref<256x1024xbf16> { 
+// CHECK-DAG: %[[c32_i64:.*]] = arith.constant 32 : i64
+// CHECK-DAG: %[[cst:.*]] = arith.constant 0.000000e+00 : bf16
+// CHECK: scf.forall (%[[ARG1:.*]], %[[ARG2:.*]]) in (8, 32) {
+// CHECK:  %[[alloc_3:.*]] = memref.alloc() {alignment = 64 : i64} : memref<32x32xbf16>
+// CHECK:  %[[temp6:.*]] = xsmm.unary.dispatch zero [32, 32, 1, 32] flags = (bcast_scalar) data_type = bf16
+// CHECK:  xsmm.unary zero(data_type = bf16, %[[temp6]], %[[cst]], %[[alloc_3]]) : (i64, bf16, memref<32x32xbf16>) -> ()
+// CHECK:  %[[temp7:.*]] = xsmm.fused_brgemm.dispatch [32, 32, 32, 32, 32, 32, 1024, 1024][add,relu]  flags = (vnni_b)  binary_flags = (bcast_col_in0)  unary_flags = (none) data_type = bf16
+// CHECK:  xsmm.fused_brgemm(data_type = bf16, %[[temp7]], %{{.*}}, %{{.*}}, %[[alloc_3]], %{{.*}}, %[[c32_i64]]) 
+// CHECK: scf.forall (%[[ARG1:.*]], %[[ARG2:.*]]) in (8, 32) {
+// CHECK:  %[[alloc_3:.*]] = memref.alloc() {alignment = 64 : i64} : memref<32x32xbf16>
+// CHECK:  %[[temp6:.*]] = xsmm.unary.dispatch zero [32, 32, 1, 32] flags = (bcast_scalar) data_type = bf16
+// CHECK:  xsmm.unary zero(data_type = bf16, %[[temp6]], %[[cst]], %[[alloc_3]]) : (i64, bf16, memref<32x32xbf16>) -> ()
```

Comment:
Why is the zero not being fused as beta=0?

Reply:
We don't fuse zeros in our current implementation yet, just brgemm, add and relu. We will need to extend the implementation for zeros. I'll update the patch with that feature.

Reply:
A bit tangential but we also don't fuse brgemm with add without relu?

Reply:
Will address both these issues in a different PR as they are in different files.